coordinator.lic

```diff
@@ -0,0 +1,478 @@
+      end
+      pause 5
+    end
+    update_box_count
+    safe_room = @settings.safe_room
+    if task[:safe_room]
+      safe_room = task[:safe_room]['id']
+    end
+    walk_to(safe_room)
+    unpause_script('sanowret-crystal') if task[:pause_sanowret]
+  end
+  
+  #
+  # Predicates
+  #  
+
+  def all_predicates
+    [
+      'hunt_done?',
+      'run_crossing_training?',
+      'pet_boxes?'
+    ]
+  end
+
+  def hunt_done?
+    # TODO: would be nice to not have to double-check predicates here.
+    !hunting_task
+  end
+  
+  def run_crossing_training?
+    @settings.crossing_training.any? { |skill| DRSkill.getxp(skill) < 28 }
+  end
+
+  def pet_boxes?
+    @num_pet_boxes > 0
+  end
+  
+  #
+  # Actions
+  #
+  def all_actions
+    [
+      'pick_boxes',
+      'train_performance',
+      'train_outfitting',
+      'train_engineering',
+      'train_forging',
+      'crossing_training',
+      'infuse_osrel',
+      'check_favors',
+    ]
+  end
+
+  def pick_boxes(task)
+    walk_to(@settings.lockpick_room_id) if @settings.lockpick_room_id
+    if @settings.waggle_sets['locksmithing']
+      wait_for_script_to_complete('buff', ['locksmithing'])
+    end
+    wait_for_script_to_complete('pick')
+    update_box_count
+  end
+
+  def infuse_osrel(task)
+    if @settings.osrel_amount && DRSpells.active_spells['Osrel Meraud']
+      infuse_om(!@settings.osrel_no_harness, @settings.osrel_amount)
+    else
+      echo '***OM IS DOWN OR NO osrel_amount***'
+      task[:deleted] = true
+    end
+  end
+
+  def check_favors(task)
+    return unless @settings.favor_goal
+    /(\d+)/ =~ bput('favor', 'You currently have \d+', 'You are not currently')
+    favor_count = Regexp.last_match(1).to_i
+    return if favor_count >= @settings.favor_goal
+
+    if /could not/ =~ bput("tap my #{@settings.favor_god} orb", 'The orb is delicate', 'I could not find')
+      wait_for_script_to_complete('favor')
+      fput('stow my orb')
+    elsif rub_orb?
+      walk_to(5865)
+      fput("get my #{@settings.favor_god} orb")
+      fput('put my orb on altar')
+      if favor_count + 1 < @settings.favor_goal
+        wait_for_script_to_complete('favor')
+        fput('stow my orb')
+      end
+    end  
+  end
+
+  def rub_orb?
+    case bput("rub my #{@settings.favor_god} orb", 'not yet fully prepared', 'lacking in the type of sacrifice the orb requires', 'your sacrifice is properly prepared')
+    when 'not yet fully prepared'
+      false
+    when 'lacking in the type of sacrifice the orb requires'
+      false
+    when 'your sacrifice is properly prepared'
+      true
+    end
+  end  
+
+  def train_performance(task)
+    return if play_song?(true)
+    
+    echo '***UNABLE TO TRAIN PERFORMANCE, REMOVING IT FROM THE TRAINING LIST***'
+    task[:deleted] = true
+  end
+
+  def train_outfitting(task)
+    return unless ensure_copper_on_hand(5000, @settings.hometown)
+    wait_for_script_to_complete('workorders', ['Tailoring'])
```

Comment:
💡 Is this still the best way to train outfitting since workorders don't pay out for f2p?